plugins/obs-qsv11/common_utils_linux.cpp

```diff
@@ -0,0 +1,279 @@
+{
+	UNUSED_PARAMETER(pthis);
+	UNUSED_PARAMETER(mid);
+	UNUSED_PARAMETER(ptr);
+	return MFX_ERR_UNSUPPORTED;
+}
+
+mfxStatus simple_unlock(mfxHDL pthis, mfxMemId mid, mfxFrameData *ptr)
+{
+	UNUSED_PARAMETER(pthis);
+	UNUSED_PARAMETER(mid);
+	UNUSED_PARAMETER(ptr);
+	return MFX_ERR_UNSUPPORTED;
+}
+
+mfxStatus simple_gethdl(mfxHDL pthis, mfxMemId mid, mfxHDL *handle)
+{
+	UNUSED_PARAMETER(pthis);
+	UNUSED_PARAMETER(mid);
+	UNUSED_PARAMETER(handle);
+	return MFX_ERR_UNSUPPORTED;
+}
+
+mfxStatus simple_free(mfxHDL pthis, mfxFrameAllocResponse *response)
+{
+	UNUSED_PARAMETER(pthis);
+	UNUSED_PARAMETER(response);
+	return MFX_ERR_UNSUPPORTED;
+}
+
+mfxStatus simple_copytex(mfxHDL pthis, mfxMemId mid, mfxU32 tex_handle,
+			 mfxU64 lock_key, mfxU64 *next_key)
+{
+	UNUSED_PARAMETER(pthis);
+	UNUSED_PARAMETER(mid);
+	UNUSED_PARAMETER(tex_handle);
+	UNUSED_PARAMETER(lock_key);
+	UNUSED_PARAMETER(next_key);
+	return MFX_ERR_UNSUPPORTED;
+}
+
+#if 0
+void ClearYUVSurfaceVMem(mfxMemId memId);
+void ClearRGBSurfaceVMem(mfxMemId memId);
+#endif
+
+// Initialize Intel Media SDK Session, device/display and memory manager
+mfxStatus Initialize(mfxIMPL impl, mfxVersion ver, MFXVideoSession *pSession,
+		     mfxFrameAllocator *pmfxAllocator, mfxHDL *deviceHandle,
+		     bool bCreateSharedHandles, bool dx9hack)
+{
+	UNUSED_PARAMETER(pmfxAllocator);
+	UNUSED_PARAMETER(deviceHandle);
+	UNUSED_PARAMETER(bCreateSharedHandles);
+	UNUSED_PARAMETER(dx9hack);
+	mfxStatus sts = MFX_ERR_NONE;
+
+	// Initialize Intel Media SDK Session
+	sts = pSession->Init(impl, &ver);
+	MSDK_CHECK_RESULT(sts, MFX_ERR_NONE, sts);
+	return sts;
+}
+
+// Release resources (device/display)
+void Release(){};
+
+void mfxGetTime(mfxTime *timestamp)
+{
+	clock_gettime(CLOCK_MONOTONIC, timestamp);
+}
+
+double TimeDiffMsec(mfxTime tfinish, mfxTime tstart)
+{
+	UNUSED_PARAMETER(tfinish);
+	UNUSED_PARAMETER(tstart);
+	//TODO, unused so far it seems
+	return 0.0;
+}
+
+extern "C" void util_cpuid(int cpuinfo[4], int level)
+{
+	__get_cpuid(level, (unsigned int *)&cpuinfo[0],
+		    (unsigned int *)&cpuinfo[1], (unsigned int *)&cpuinfo[2],
+		    (unsigned int *)&cpuinfo[3]);
+}
+
+struct vaapi_device {
+	int fd;
+	VADisplay display;
+	const char *driver;
+};
+
+void vaapi_open(char *device_path, struct vaapi_device *device)
+{
+	int fd = open(device_path, O_RDWR);
+	if (fd < 0) {
+		return;
+	}
+
+	VADisplay display = vaGetDisplayDRM(fd);
+	if (!display) {
+		close(fd);
+		return;
+	}
+
+	// VA-API is noisy by default.
+	vaSetInfoCallback(display, nullptr, nullptr);
+	vaSetErrorCallback(display, nullptr, nullptr);
+
+	int major;
+	int minor;
+	if (vaInitialize(display, &major, &minor) != VA_STATUS_SUCCESS) {
+		vaTerminate(display);
+		close(fd);
+		return;
+	}
+
+	const char *driver = vaQueryVendorString(display);
+	if (strstr(driver, "Intel i965 driver") != nullptr) {
+		blog(LOG_ERROR,
+		     "Legacy intel-vaapi-driver detected, incompatible with QSV");
```

Comment:
Is this really needs to be a error ?
A warning might be better because it only shows that the adapter use the old VAAPI driver, which mean that the adapter is too old for the plugin.

Reply:
It does not mean the card is too old, a fairly wide selection of chips sit in the overlap of support between both drivers where users could have this functionality if they used the newer driver.